sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/SessionReceiverManager.cs

```diff
@@ -279,15 +269,18 @@ public override async Task ReceiveAndProcessMessagesAsync(CancellationToken proc
                     }
                 }
                 catch (ServiceBusException ex)
-                when (ex.Reason == ServiceBusFailureReason.ServiceTimeout)
+                    when (ex.Reason == ServiceBusFailureReason.ServiceTimeout)
                 {
                     // these exceptions are expected when no messages are available
                     // so simply return and allow this to be tried again on next thread
                     return;
                 }
+                finally
+                {
+                    _concurrentAcceptSessionsSemaphore.Release();
```

Comment:
The class state mutation here makes me a bit uncomfortable.   Is there any way that we can have whomever owns semaphore acquisition also have responsibility for release?

Reply:
Unfortunately no, because we need to release the lock as soon as a session has been accepted or we determine that we don't need to accept a new session.

Reply:
Open to other ideas though.

Reply:
The best that I have is to comment around it.